File: src/main/java/fr/anthonus/commands/user/CursedCommand.java
package fr.anthonus.commands.user;

import fr.anthonus.commands.Command;
import fr.anthonus.logs.LOGs;
import fr.anthonus.logs.logTypes.DefaultLogType;
import fr.anthonus.utils.Utils;
import fr.anthonus.utils.api.RedditAPI;
import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
import net.dv8tion.jda.api.utils.FileUpload;

import java.io.IOException;

public class CursedCommand extends Command {
    public CursedCommand(SlashCommandInteractionEvent event) {
        super(event);

        LOGs.sendLog("Commande /cursed initialisée", DefaultLogType.COMMAND);
    }

    @Override
    public void run() {
        currentEvent.deferReply().queue();

        String imgURL = RedditAPI.getBlursedImageURL("blursedimages");
        try {
            byte[] imageBytes = Utils.downloadImageToByteArray(imgURL);
            String fileName = imgURL.substring(imgURL.lastIndexOf("/") + 1);

            currentEvent.getHook().sendFiles(FileUpload.fromData(imageBytes, fileName)).queue();
        } catch (IOException e) {
            EmbedBuilder embed = new EmbedBuilder();

            embed.setTitle(":x: ERREUR :x:");
            embed.setDescription("Une erreur est survenue lors du téléchargement de l'image.");

            embed.setColor(0xFF0000);

            currentEvent.getHook().sendMessageEmbeds(embed.build()).queue();
        }
    }
}

File: src/main/kotlin/discord/commands/global/CursedCommand.kt
package discord.commands.global

import api.RedditApi
import discord.commands.Command
import helpers.EmbedHelper
import helpers.LogsHelper
import helpers.NetHelper
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import net.dv8tion.jda.api.utils.FileUpload

class CursedCommand : Command() {
    override val name = "cursed"
    override val description = "Envoie une image aléatoire du subreddit r/blursed"

    override fun executeBody(event: SlashCommandInteractionEvent) {
        event.deferReply().queue()

        val imageUrl = RedditApi.getRandomImage("blursedimages")
        if (imageUrl == null) {
            event.hook.editOriginalEmbeds(
                EmbedHelper.createEmbed(
                    type = EmbedHelper.Type.ERROR,
                    title = "une erreur est survenue lors de la récupération de l'URL"
                )
            ).queue()

            LogsHelper.log.error(
                "Error while retrieving reddit image url",
                Exception("Error while retrieving reddit image url")
            )

            return
        }

        val fileName = imageUrl.substring(imageUrl.lastIndexOf("/") + 1)
        val imageData = FileUpload.fromData(NetHelper.downloadImage(imageUrl), fileName)

        event.hook.sendFiles(imageData)
            .queue(
                {
                    LogsHelper.success(event, "Cursed reddit image sent successfully")
                },
                { err ->
                    LogsHelper.failure(event, "Cursed reddit image not sent", err)
                }
            )
    }
}